src/main/java/shortestpath/pathfinder/PathfinderConfig.java

```diff
@@ -48,17 +54,23 @@ public class PathfinderConfig {
     private int recalculateDistance;
     private Map<Quest, QuestState> questStates = new HashMap<>();
 
-    public PathfinderConfig(CollisionMap map, Map<WorldPoint, List<Transport>> transports, Client client,
+    public PathfinderConfig(SplitFlagMap mapData, Map<WorldPoint, List<Transport>> transports, Client client,
                             ShortestPathConfig config, ShortestPathPlugin plugin) {
-        this.map = map;
+        this.mapData = mapData;
+        this.map = ThreadLocal.withInitial(() -> new CollisionMap(this.mapData));
         this.allTransports = transports;
         this.transports = new HashMap<>();
+        this.transportsPacked = new HashMap<>();
         this.client = client;
         this.config = config;
         this.plugin = plugin;
         refresh();
     }
 
+    public CollisionMap getMap() {
+        return map.get();
+    }
+
     public void refresh() {
         calculationCutoff = Duration.ofMillis(config.calculationCutoff() * Constants.GAME_TICK_LENGTH);

```

Comment:
Since you changed from using Instant to milliseconds I would want to change calculationCutoff from a Duration to a long containing milliseconds. I guess just multiply with 1000 here.